ctapipe/io/simteleventsource.py

```diff
@@ -414,6 +414,13 @@ def _generate_events(self):
                 if r0.num_trig_pix > 0:
                     r0.trig_pix_id = pixel_lists[0]["pixel_list"]
 
+                # get time_shift from laser calibration
+                time_calib = array_event["laser_calibrations"][tel_id]["tm_calib"]
+                pix_index = np.arange(n_pixels)
+
+                dl1_calib = data.calibration.tel[tel_id].dl1
+                dl1_calib.time_shift = time_calib[r1.selected_gain_channel, pix_index]
```

Comment:
why is pix_index needed?

Reply:
Same reason as in the R1 Calibrator. we have a (n_gain, n_pixels) array of time shifts and need to select for each pixel the correct time shift for the selected gain channel.

If you have a better solution, I'll be glad to adapt it also in the other code (which I think you wrote?)

Reply:
Ah right, yeh. No I don't have a better solution